cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowActionContext.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.workflow;
+
+import co.cask.cdap.api.RuntimeContext;
+import co.cask.cdap.api.ServiceDiscoverer;
+import co.cask.cdap.api.TxRunnable;
+import co.cask.cdap.api.data.DatasetContext;
+import co.cask.cdap.api.dataset.Dataset;
+
+import java.util.Map;
+
+/**
+ * Defines an interface for the context of Workflow Actions
+ */
+public interface WorkflowActionContext extends RuntimeContext, ServiceDiscoverer, DatasetContext {
+
+  /**
+   * Returns the specification used to configure {@link WorkflowAction} bounded to this context.
+   */
+  WorkflowActionSpecification getSpecification();
+
+  /**
+   * Execute a set of operations on datasets via a {@link TxRunnable} that are committed as a single transaction.
+   * @param txRunnable the runnable to be executed in the transaction
+   */
+  void execute(TxRunnable txRunnable);
+
+  /**
+   * @return the instance id
+   */
+  int getInstanceId();
+
+  /**
+   * @return the datasets used by this workflow action
+   */
+  Map<String, Dataset> getDatasets();
```

Comment:
Since the execute() method is not needed, I don't think there is need for this method. Also, even we really need it, I don't think it should a public API.